13_browser.md

```diff
@@ -1,458 +1,263 @@
-web page it was embedded in.
+然而，万维网的魅力就在于你浏览网页的时候不必((信任))这些访问的页面。这也正是为什么浏览器极大地限制了 JavaScript 程序可以做的事情：它不能读取你计算机上的文件，或是修改任何与该程序嵌入的网页无关的事物。
 
 {{index isolation}}
 
-Isolating a programming environment in this way is called
-_((sandbox))ing_, the idea being that the program is harmlessly
-playing in a sandbox. But you should imagine this particular kind of
-sandbox as having a cage of thick steel bars over it so that the
-programs playing in it can't actually get out.
+以这样的方式将编程环境分离而出，我们称之为 _((沙箱))化_，意思是程序像是在一个沙箱中无害地进行玩闹。但是你应该将这种沙箱想像成被铜墙铁壁所包围，所以事实上在其中玩闹的程序是出不去的。
 
-The hard part of sandboxing is allowing the programs enough room to be
-useful yet at the same time restricting them from doing anything
-dangerous. Lots of useful functionality, such as communicating with
-other servers or reading the content of the copy-paste ((clipboard)),
-can also be used to do problematic, ((privacy))-invading things.
+沙箱化的难点在于，在给程序足够的空间使其能发挥作用的同时，也得限制它们使其不能做出危险的事。有很多有用的功能 —— 比如说与别的服务器通信，或是读取复制粘贴的((剪贴板))的内容 —— 这些功能也会被用来做糟糕的、侵犯((隐私))的事情。
 
 {{index leak, exploit, security}}
 
-Every now and then, someone comes up with a new way to circumvent the
-limitations of a ((browser)) and do something harmful, ranging from
-leaking minor private information to taking over the whole machine
-that the browser runs on. The browser developers respond by fixing the
-hole, and all is well again—until the next problem is discovered, and
-hopefully publicized, rather than secretly exploited by some
-government agency or ((mafia)).
+每过一段时间，总有人会想出新的办法去规避((浏览器))的限制，并且做出有害的事情，从泄漏少量的私人信息到控制浏览器运行其上的整个机器不等。浏览器开发者则会以修补漏洞来回应，然后岁月静好 —— 直到下一个漏洞被发现，最好是公之于众的，而不是被一些政府机构或((黑帮))私下利用。
 
-## Compatibility and the browser wars
+## 兼容性与浏览器大战
 
 {{index Microsoft, "World Wide Web"}}
 
-In the early stages of the Web, a browser called ((Mosaic)) dominated
-the market. After a few years, the balance shifted to
-((Netscape)), which was then, in turn, largely supplanted by
-Microsoft's ((Internet Explorer)). At any point where a single
-((browser)) was dominant, that browser's vendor would feel entitled to
-unilaterally invent new features for the Web. Since most users used
-the most popular browser, ((website))s would simply start using those
-features—never mind the other browsers.
-
-This was the dark age of ((compatibility)), often called the
-_((browser wars))_. Web developers were left with not one unified Web
-but two or three incompatible platforms. To make things worse, the
-browsers in use around 2003 were all full of ((bug))s, and of course
-the bugs were different for each ((browser)). Life was hard for people
-writing web pages.
+在万维网的早期，一个名为 ((Mosaic)) 的浏览器统治着市场。过了短短几年之后，天平移向了((Netscape))，而它随后又被微软的((Internet Explorer))占据了大部分市场。无论何时，每当有一个((浏览器))统治了市场，该浏览器厂商便会想要独自为万维网发明新的特性。由于大多数用户使用最流行的浏览器，((网站))便会开始使用这些特性 —— 而不管别的浏览器是否兼容。
+
+这是((兼容性))的黑暗时代，通常被称为 _((浏览器大战))_。网页开发者并没有一个统一的万维网，摆在他们面前的是两到三个互不兼容的平台。更糟糕的是，2003 年左右的浏览器有很多 ((bug))，当然每个((浏览器))的 bug 还都不一样。那时候网页开发者的生活糟透了。
```

Comment:
bug should be pluralized

Reply:
No problem. Btw, It is good to see you here on Github again :)